script.js

```diff
@@ -98,19 +134,175 @@ var getDefaultWinLossMessage = function () {
   );
 };
 
-// Check whether player draws with computer
+/**
+ * Check whether player draws with computer objects
+ * @param {string} playerObject
+ * @param {string} computerObject
+ */
 var doesPlayerDrawWComputer = function (playerObject, computerObject) {
+  // determine draw conditions for 'reversed sps'. Return a boolean
+  if (mode == REVERSED_SPS) {
+    return (
+      (playerObject == REVERSED_SCISSORS && computerObject == SCISSORS) ||
+      (playerObject == REVERSED_PAPER && computerObject == PAPER) ||
+      (playerObject == REVERSED_STONE && computerObject == STONE)
+    );
+  }
+  // determine draw conditions for regular sps. Return a boolean
+  return playerObject == computerObject;
+};
+
+/**
+ * Check whether player's inputs are valid
+ * @param {string} userInput
+ */
+var validateObjectInput = function (userInput) {
+  //if user is playing reversed SPS, validate  input accordingly
+  if (mode == REVERSED_SPS) {
+    return (
+      userInput == REVERSED_SCISSORS ||
+      userInput == REVERSED_PAPER ||
+      userInput == REVERSED_STONE
+    );
+  }
+  // if use is playing regular SPS, validate input accordingly
+  return userInput == SCISSORS || userInput == PAPER || userInput == STONE;
+};
+
+/**
+ * Check whether player's input on the mode is valid
+ */
+var validateModeInput = function (userInput) {
   return (
-    playerObject == computerObject ||
-    (playerObject == REVERSED_SCISSORS && computerObject == SCISSORS) ||
-    (playerObject == REVERSED_PAPER && computerObject == PAPER) ||
-    (playerObject == REVERSED_STONE && computerObject == STONE)
+    userInput == REGULAR ||
+    userInput == REVERSED_SPS ||
+    userInput == KOREAN_SPS ||
+    userInput == COM_V_COM
   );
 };
 
 /**
- * Play SPS with user input, return game result.
+ * Create a string message saying that the user's  input is invalid
+ */
+var createResponseForInvalidInput = function () {
+  // if the mode is 'reversed sps', return the relevant instructions
+  if (mode == REVERSED_SPS) {
+    return (
+      'Please input 1 of ' +
+      REVERSED_SCISSORS +
+      ', ' +
+      REVERSED_PAPER +
+      ', or ' +
+      REVERSED_STONE +
+      ' to play the game.'
```

Comment:
Should we use template literals to make this more concise? Seems very clunky and not nice code like this, potentially making the overall code harder to read. @jeremylim91 what do you think? CC @awongh